File: 4way/problem006/solution.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isArraySpecial = function(nums) {
    if(nums.length == 1) return true
    let prev = nums[0] % 2 == 0
    for(let i = 1; i < nums.length; i++){
        if(prev == (nums[i] % 2 == 0)) return false
        prev = !prev
    }
    return true
};

File: 4way/problem006/solution.c
bool isArraySpecial(int* nums, int numsSize) {
    if(numsSize == 1) return true;
    bool prev = nums[0] % 2 == 0;
    for(int i = 1; i < numsSize; i++){
        if(prev == (nums[i] % 2 == 0)) return false;
        prev = !prev;
    }
    return true;
}
